### PR TITLE
Feature add pip deps

### PR DESCRIPTION
**Pull request by bvanessen**
Cleanup how external Python dependencies are installed via PIP.

### CODE REVIEW

File: docs/running_lbann.rst [hunk 1]
```diff
@@ -7,6 +7,13 @@
 Running LBANN
 ============================================================
 
+------------------------------------------------
+Sanity check with a simple test using LeNet
+------------------------------------------------
+
+In the :ref:`Quick Start <test-lbann-install` guide there is an
```
Comment (benson31) [inline docs/running_lbann.rst:14]:
```suggestion
In the :ref:`Quick Start <test-lbann-install>` guide there is an
```

File: docs/running_lbann.rst [hunk 2]
```diff
@@ -453,3 +430,48 @@ The LBANN Protobuf format is defined in `src/proto/lbann.proto
...
 fields).
+
+------------------------------------------------
+Setup of LBANN for manual CMake build (advanced)
+------------------------------------------------
+
+If LBANN was compiled and installed directly using CMake there is a
+bit more work that is required to ensure that the Python front end
+will work.  The :python:`lbann` Python package is installed as part of
+the LBANN build process.However, it is necessary to update the
```
Comment (benson31) [inline docs/running_lbann.rst:441]:
```suggestion
the LBANN build process. However, it is necessary to update the
```

File: docs/running_lbann.rst [hunk 3]
```diff
@@ -453,3 +430,48 @@ The LBANN Protobuf format is defined in `src/proto/lbann.proto
...
+.. warning:: The above will *not* work if LBANN has been built with
+             :bash:`scripts/build_lbann_lc.sh` or with the Spack
+             developer build process.
+
++ LBANN includes a modulefile that updates :bash:`PYTHONPATH`:
+
+.. code-block:: bash
+
+    module use <install directory>/etc/modulefiles
+    module load lbann-<version>
```
Comment (benson31) [inline docs/running_lbann.rst:461]:
```suggestion
    module load lbann/<version>
```

**Review by benson31 — COMMENTED** [review]
Initial comments

**Review by benson31 — APPROVED** [review]
LGTM

File: applications/ATOM/build_lbann_atom_user.sh [hunk 4]
```diff
@@ -1,8 +1,10 @@
-LBANN_REPO_PATH="https://github.com/LLNL/lbann/raw/develop"
+LBANN_REPO_PATH="https://github.com/bvanessen/lbann/raw/feature_add_pip_deps"
+#LBANN_REPO_PATH="https://github.com/LLNL/lbann/raw/develop"
```
Comment (timmoon10) [inline applications/ATOM/build_lbann_atom_user.sh:2]:
```suggestion
LBANN_REPO_PATH="https://github.com/LLNL/lbann/raw/develop"
```

File: scripts/.build_lbann_as_user.sh [hunk 5]
```diff
@@ -1,7 +1,9 @@
-LBANN_REPO_PATH="https://github.com/LLNL/lbann/raw/develop"
+#LBANN_REPO_PATH="https://github.com/LLNL/lbann/raw/develop"
+LBANN_REPO_PATH="https://github.com/bvanessen/lbann/raw/feature_add_pip_deps"
```
Comment (timmoon10) [inline scripts/.build_lbann_as_user.sh:2]:
```suggestion
LBANN_REPO_PATH="https://github.com/LLNL/lbann/raw/develop"
```

**Review by timmoon10 — APPROVED** [review]
Overall looks good. I suspect there is some debugging code that should be reverted.

**Pull request merged**